<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>748</width>
    <height>459</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Tab 1</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="9" column="0">
          <widget class="QPushButton" name="gen">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Generate</string>
           </property>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QDoubleSpinBox" name="cost">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="maximum">
            <double>999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Gamma</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>End temp</string>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QPushButton" name="problem">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Problem</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QDoubleSpinBox" name="endTemp">
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>9999.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Iterations</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Alpha</string>
           </property>
          </widget>
         </item>
         <item row="10" column="0">
          <widget class="QPushButton" name="run">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Run</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QDoubleSpinBox" name="alpha">
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>0.990000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.500000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QDoubleSpinBox" name="beta">
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>0.990000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QSpinBox" name="iterations">
           <property name="maximum">
            <number>9999</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QDoubleSpinBox" name="gamma">
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>0.990000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QDoubleSpinBox" name="initTemp">
           <property name="frame">
            <bool>true</bool>
           </property>
           <property name="readOnly">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <double>1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999.000000000000000</double>
           </property>
           <property name="value">
            <double>1000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QPushButton" name="toolchain">
           <property name="text">
            <string>Toolchain</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Beta</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Init temp</string>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QLabel" name="label_status">
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCustomPlot" name="plot" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <attribute name="title">
       <string>Page</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QCustomPlot" name="widget_2" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Tab 2</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,0,0,0">
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="font">
              <font>
               <pointsize>15</pointsize>
               <weight>75</weight>
               <italic>false</italic>
               <bold>true</bold>
               <underline>false</underline>
               <strikeout>false</strikeout>
               <kerning>true</kerning>
              </font>
             </property>
             <property name="mouseTracking">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Tool route:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBox_2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="layoutDirection">
              <enum>Qt::LeftToRight</enum>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="ToolPathWidget" name="widget" native="true"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ToolPathWidget</class>
   <extends>QWidget</extends>
   <header>toolpathwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>toolchain</tabstop>
  <tabstop>beta</tabstop>
  <tabstop>alpha</tabstop>
  <tabstop>endTemp</tabstop>
  <tabstop>problem</tabstop>
  <tabstop>run</tabstop>
  <tabstop>initTemp</tabstop>
  <tabstop>gamma</tabstop>
  <tabstop>iterations</tabstop>
  <tabstop>cost</tabstop>
  <tabstop>gen</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>toolchain</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>selectToolchain()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>73</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>152</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>problem</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>selectProblem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>68</x>
     <y>335</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>run</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>419</y>
    </hint>
    <hint type="destinationlabel">
     <x>178</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gen</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>generate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>379</y>
    </hint>
    <hint type="destinationlabel">
     <x>136</x>
     <y>384</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>selectToolchain()</slot>
  <slot>selectProblem()</slot>
  <slot>run()</slot>
  <slot>generate()</slot>
 </slots>
</ui>
